x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +241,147 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
+	params := suite.App.GAMMKeeper.GetParams(suite.Ctx)
+	testAccount := suite.TestAccs[0]
+
+	// get raw pool creation fee(s) as DecCoins
+	poolCreationFeeDecCoins := sdk.DecCoins{}
+	for _, coin := range params.PoolCreationFee {
+		poolCreationFeeDecCoins = poolCreationFeeDecCoins.Add(sdk.NewDecCoin(coin.Denom, coin.Amount))
+	}
+
+	_, err := balancer.NewBalancerPool(1, defaultPoolParams, defaultPoolAssets, defaultFutureGovernor, time.Now())
+	require.NoError(suite.T(), err)
+
+	_, err = stableswap.NewStableswapPool(1, defaultStableSwapPoolParams, defaultStableSwapPoolAssets, defaultScalingFactor, "", defaultFutureGovernor)
+	require.NoError(suite.T(), err)
+
+	tests := []struct {
+		name        string
+		pool        func() swaproutertypes.PoolI
+		expectPass  bool
+		expectPanic bool
+	}{
+		{
+			name: "initialize balancer pool with default assets",
+			pool: func() swaproutertypes.PoolI {
+				balancerPool, err := balancer.NewBalancerPool(
+					defaultPoolId,
+					defaultPoolParams,
+					defaultPoolAssets,
+					"",
+					time.Now(),
+				)
+				require.NoError(suite.T(), err)
+				return &balancerPool
+			},
+			expectPass:  true,
+		},
+		{
+			name: "initialize stableswap pool with default assets",
+			pool: func() swaproutertypes.PoolI {
+				stableswapPool, err := stableswap.NewStableswapPool(
+					defaultPoolId,
+					defaultPoolParamsStableSwap,
+					defaultStableSwapPoolAssets,
+					defaultScalingFactor,
+					"",
+					defaultFutureGovernor,
+				)
+				require.NoError(suite.T(), err)
+				return &stableswapPool
+			},
+			expectPass:  true,
+		},
+		{
+			name: "initialize a CL pool which cause panic",
+			pool: func() swaproutertypes.PoolI {
+				clPool, err := clmodel.NewConcentratedLiquidityPool(
+					defaultPoolId,
+					ETH,
+					USDC,
+					defaultTickSpacing,
+				)
+				require.NoError(suite.T(), err)
+				return &clPool
+			},
+			expectPanic: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		suite.Run(test.name, func() {
+			osmoassert.ConditionalPanic(suite.T(), test.expectPanic, func() {
+				suite.SetupTest()
+
+				gammKeeper := suite.App.GAMMKeeper
+				bankKeeper := suite.App.BankKeeper
+				poolIncentivesKeeper := suite.App.PoolIncentivesKeeper
+
+				// fund sender test account
+				var sender sdk.AccAddress
+				sender = testAccount
+
+				// initializePool with a poolI
+				err = gammKeeper.InitializePool(suite.Ctx, test.pool(), sender)
+
+				if test.expectPass {
+					suite.Require().NoError(err, "test: %v", test.name)
+
+					// check to make sure new pool exists and has minted the correct number of pool shares
+					pool, err := gammKeeper.GetPoolAndPoke(suite.Ctx, defaultPoolId)
+					suite.Require().NoError(err, "test: %v", test.name)
+					suite.Require().Equal(types.InitPoolSharesSupply.String(), pool.GetTotalShares().String(),
+						fmt.Sprintf("share token should be minted as %s initially", types.InitPoolSharesSupply.String()),
```

Comment:
nit: This auto-expands when you use %s format
```suggestion
						fmt.Sprintf("share token should be minted as %s initially", types.InitPoolSharesSupply),
```